README.md

```diff
@@ -1,133 +1,114 @@
-`numpy` is used in the function to serialise the values of the hash into an
-`ndarray` with a shape `(1, 30)` after which it's stored as a [RedisAI tensor](https://oss.redislabs.com/redisai/intro/#using-redisai-tensors).
-
-![Updating reference data](./flow1.png "Updating reference data")
-
-
-### Flow 2: Transaction scoring
-When a new transaction happens, it needs to be evaluated if it's fraudulent or not.  The ML/DL models take two inputs for this, the relevant reference data and the new transaction details.  RedisGears will orchestrate the fetching of the relevant reference data and creating a single input tensor out of it via a Torch Script.  Afterwards it will execute several inferences using TensorFlow models hosted inside Redis and combine the results to reply the transaction scoring result.
-
-![High level architecture](./flow2.png "High level architecture")
-
-An input tensor is set in the keyspace with the shape of `(1, 30)`. A second RedisGears function is triggered with as input
-- the keyname holding the new transaction details tensor
-- a time range, represented by two timestamps.
-
-The function then executes a range query over the sorted set ([`ZRANGEBYSCORE`](https://redis.io/commands/zrangebyscore)) and retrieves a list of hash names (recall that each hash has a corresponding tensor as described in Flow 1).
-
-![Gears<->Redis data gathering](./flow3.png "Gears<->Redis data gathering")
-
-From this list it extracts a list of tensors from the keyspace and sends it to a [Torch script](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/script.torch). This torch script creates a
-new tensor with the shape `(1, 256)`, by concatenating the tensors and either pad the remaining space or trimming it.
-The new tensor is the reference data for the models which expect a reference data with shape of `(1, 256)` and transaction details tensor with the shape of `(1, 30)`. Once the inferencing of the models are done, the function uses `numpy` to aggregate the results and save them to a tensor
-with shape `(1, 2)` that contains the probability for the transaction to be a fraud.
-
-![Gears<->AI execution](./flow4.png "Gears<->AI execution")
+This demo showcases the advantage of **data locality** during transaction scoring, by using [RedisAI](https://oss.redislabs.com/redisai/) module.
+It simulates a fraud-detection app which is based on ML model prediction in real-time. When a new transaction is executed, it triggers a flow whose output is the probability that this transaction is fraudulent. The prediction is based on both the given transaction and some reference data stored in Redis, while *the entire flow* is done within Redis as well.   
```

Comment:
This demo simulates a fraud-detection app, which relies on a ML model returning the probability that a given transaction is fraudulent. The model uses transaction data and reference (feature) data previously stored in Redis for the customers. The entire flow is executed within redis